src/cattr/_compat.py

```diff
@@ -73,6 +71,21 @@ def is_bare(type):
             or args == bare_mutable_seq_args
         )
 
+    @lru_cache(32)
+    def _get_class_namespace(class_: type) -> Dict[str, Any]:
+        return vars(sys.modules[class_.__module__])
+
+    @lru_cache(32)
+    def get_type(parent_class: type, type_: Union[type, str]) -> type:
```

Comment:
there's not really a reason to avoid `type_` as an argument name; it can be plain `type` as the other functions in this file also do.

also the `Union[type, str]` can probably become `Union[Type, str]` with a capitalized `Type`. this needs a `from typing import Type` at the top of the file.

see https://docs.python.org/3/library/typing.html#typing.Type, especially

> `Type[Any]` is equivalent to `Type` which in turn is equivalent to `type`, which is the root of Python’s metaclass hierarchy.

Reply:
the checks done by flake8 (and especially by semi-random plugins) are by their very nature opinionated so that does hardly count as absolute truth.

sure, ideally there would be no clashing names.

however, it is also ‘better to avoid’ building APIs with clunky names (especially when the API is not just for internal use) :wink: i can think of many cases in which arguments (especially keyword arguments) with names that are also built-ins actually _do_ seem the best name. some examples: `type=...`, `min=...`, `max=...`, `filter=...`, `all=...`, `set=...`, `open=...`, `round=...`